Source/WebKit/NetworkProcess/soup/NetworkDataTaskSoup.cpp

```diff
@@ -521,7 +521,11 @@ void NetworkDataTaskSoup::dispatchDidReceiveResponse()
 
             break;
         case PolicyAction::Ignore:
-            clearRequest();
+            if (isDownload()) {
+                invalidateAndCancel();
+                dispatchDidCompleteWithError(downloadCancelledByUserError(m_response));
```

Comment:
Looking at what Cocoa port does I believe it may make sense to run this code unconditionally when the policy action is ignore.

Reply:
I know it's been a while, but any chance you remember why you suggested this? `NetworkDataTaskCocoa::didReceiveResponse` doesn't look anything like this.